app/controllers/users_controller.rb

```diff
@@ -1,9 +1,10 @@
 # frozen_string_literal: true
 
 class UsersController < ApplicationController
-  before_action :authenticate_user!, except: [:select_nearby_delegate, :acknowledge_cookies]
+  before_action :authenticate_user!, except: [:select_nearby_delegate, :acknowledge_cookies, :anonymize]
```

Comment:
Why are you skipping authentication for the `anonymize` step?
It most definitely should only be available to authenticated users